### PR TITLE
Add a note on shared worker lifetime

### LINKED ISSUE #23439: SharedWorker lifetime
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers

### What specific section or headline is this issue about?

When does a SharedWorker end?

### What information was incorrect, unhelpful, or incomplete?

The article explains how a SharedWorker is started, but it doesn't tell anything about when it ends. Can it run forever? Also in the background, after all pages from the same origin were closed? If not, when is it stopped and will it know or will it simply disappear?

### What did you expect to see?

Explanation of the SharedWorker lifetime

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/web_workers_api/using_web_workers`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/web_workers_api/using_web_workers/index.md
* Last commit: https://github.com/mdn/content/commit/2a0f377afe6b03266af2597be941365bea9bc9b6
* Document last modified: 2022-10-30T06:11:44.000Z

</details>

### PR DESCRIPTION
**Pull request by ambujsahu81**
### Description

- Adds a small note on shared worker lifetime.

### Motivation

Added note will help anyone who want to know more about worker lifetime. And notes also explain when the shared worker is terminated

### Additional details

Reference used to create note :- 1) [shared_worker_life_time]( https://html.spec.whatwg.org/multipage/workers.html#the-worker's-lifetime)
                    2) [Initial_text_about_sharedworker_creation](https://html.spec.whatwg.org/dev/workers.html#infrastructure-2)

### Related issues and pull requests

Fixes  [#12312 ](https://github.com/mdn/content/issues/23439)

### CODE REVIEW

File: files/en-us/web/api/sharedworker/index.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ The following code snippet shows creation of a `SharedWorker` object using the {
 const myWorker = new SharedWorker('worker.js');
 ```
 
+> **Note:** Once Shared worker is created any script running in the same origin can obtain a reference to that worker and communicate with it. And shared worker will be alive as long as its WorkerGlobalScope's owner set (a set of Document and WorkerGlobalScope objects) is not empty. (e.g., There is any live page holding a reference to it (through new SharedWorker())). To read more about shared worker lifetime see [10.2.3 The worker's lifetime](https://html.spec.whatwg.org/multipage/workers.html#the-worker's-lifetime).
```
Comment (Josh-Cena) [inline files/en-us/web/api/sharedworker/index.md:54]:
```suggestion
> **Note:** Once a shared worker is created, any script running in the same origin can obtain a reference to that worker and communicate with it. The shared worker will be alive as long as its global scope's owner set (a set of `Document` and `WorkerGlobalScope` objects) is not empty. (e.g., there is any live page holding a reference to it, maybe through `new SharedWorker()`). To read more about shared worker lifetime, see [10.2.3 The worker's lifetime](https://html.spec.whatwg.org/multipage/workers.html#the-worker's-lifetime).
```

**Comment by ambujsahu81** [comment]
Hi , is there any suggestion or modification that i can add to the note to describe shared worker lifetime in a better way?

File: files/en-us/web/api/sharedworker/index.md [hunk 2]
```diff
@@ -51,6 +51,8 @@ The following code snippet shows creation of a `SharedWorker` object using the {
 const myWorker = new SharedWorker('worker.js');
 ```
 
+> **Note:** Once a shared worker is created, any script running in the same origin can obtain a reference to that worker and communicate with it. The shared worker will be alive as long as its global scope's owner set (a set of `Document` and `WorkerGlobalScope` objects) is not empty. (e.g., there is any live page holding a reference to it, maybe through `new SharedWorker()`). To read more about shared worker lifetime, see [10.2.3 The worker's lifetime](https://html.spec.whatwg.org/multipage/workers.html#the-worker's-lifetime).
```
Comment (sideshowbarker) [inline files/en-us/web/api/sharedworker/index.md:54]:
```suggestion
> **Note:** Once a shared worker is created, any script running in the same origin can obtain a reference to that worker and communicate with it. The shared worker will be alive as long as its global scope's owner set (a set of `Document` and `WorkerGlobalScope` objects) is not empty (for example, if there is any live page holding a reference to it, maybe through `new SharedWorker()`). To read more about shared worker lifetime, see [The worker's lifetime](https://html.spec.whatwg.org/multipage/workers.html#the-worker's-lifetime) section of the HTML specification.
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
@ambujsahu81

> Hi , is there any suggestion or modification that i can add to the note to describe shared worker lifetime in a better way?

What you wrote is great — thanks much, and sorry for the delay in getting it merged

**Pull request merged**